### ISSUE TITLE
Skills Advance

### ISSUE THREAD

**Issue by killer000** [aura-project/aura#518]
The button isn't grayed out when you don't have enough AP for the skill

**Comment by exectails** [comment]
Should it be? And if so, was it always like that? I think I remember clicking Advance just to realize I don't have the amount of AP shown in the confirmation window o.o

**Comment by Pegwald** [comment]
I believe the text in "Advance" is greyed out if you don't have enough AP. Not entirely sure myself, but I remember you got an idea whether or not you could afford the skill or not

**Comment by killer000** [comment]
Yeah you could still click on it to see how much AP you have, but it shouldn't be lit up unless you have the AP for it.

**Comment by exectails** [comment]
I've actually left a comment about this in the source code:
```csharp
/// MaxRank: This probably regulates the Advance button, officials set
///   it to the next valid rank (current one if you can't advance yet).
```
So we'll basically need an AP change event on the creature so we can update the skill's MaxRank accordingly.

**Comment by exectails** [comment]
Well, there's one problem, I didn't realize we're missing the update packet o,o" We'll need someone to get that from officials. Either how you spend AP to make buttons become grayed out, or gain AP to make them "available".

**Comment by exectails** [comment]
I was just on NA and thought to myself "might just as well check my characters to see if I can get this myself"... yea, right...
![](http://puu.sh/sVpEQ/6f4ae965e6.jpg)
I hate you, devCAT >_>

**Comment by Pegwald** [comment]
I also seem to remember this, when you finally level up so you can advance that skill you saved so much ap for, you will get a reminder appearing on the screen for a moment that " x can now be upgraded" (once you hit enough ap required ofc). This happens with pretty much any skill that is available for advancement once you pass their required ap.

**Comment by exectails** [comment]
Oh~ you're right, I do remember that.
